test/inline-compare/compare.py

```diff
@@ -150,7 +151,12 @@ def vulnerabilities(self):
             if not INCLUDE_SEVERITY:
                 severity = NO_COMPARE_VALUE
 
-            metadata[package.type][package] = Metadata(version=entry["artifact"]["version"], severity=severity)
+            # engine doesn't capture epoch info, so we cannot use it during comparison
+            version = entry["artifact"]["version"]
```

Comment:
these fields are probably mandatory, but I generally prefer the "get" approach to avoid a KeyError (unless that's what you're wanting here)
```
entry.get("artifact", {}).get("version")
```

Reply:
agreed, in fact I usually bring in a [small `dig` utility](https://github.com/anchore/anchore-engine/blob/master/anchore_engine/analyzers/utils.py#L1440) for safe access that makes things more readable than the vanilla approach.

This script will be removed from the repo soon once grype is integrated into engine in 3.2, so I won't incorporate this comment go around.